src/components/main/files/upload/mod.rs

```diff
@@ -45,16 +52,21 @@ pub fn Upload<'a>(cx: Scope<'a, Props<'a>>) -> Element<'a> {
                                     let local_path = Path::new(&file_path).to_string_lossy().to_string();
                                     let mut filename_to_save = filename.clone();
                                     let mut count_index_for_duplicate_filename = 1;
-                                    let mut write_storage = file_storage.write();
+
                                     loop {
-                                        match write_storage.put(&filename_to_save, &local_path).await {
-                                            Ok(_) => {
-                                                println!("{:?} file uploaded", &filename_to_save); 
+                                        match file_storage.put(&filename_to_save, &local_path).await {
```

Comment:
Not required to change but more of a notation, howeveri do have `Constellation::put_stream` which would save a stream of data. This function does have a progression stream which would could tell one how long it would take to upload/save, which may be suitable for some type of notification in the future. Other functions will be refactored to have that but for now only `put_stream` supports it.